CMakeLists.txt

```diff
@@ -16,495 +17,218 @@ cmake_minimum_required (VERSION 2.8.12)
-    target_include_directories(${TARGET_NAME} PRIVATE ${PORT_DIR})
-    target_include_directories(${TARGET_NAME} SYSTEM PRIVATE ${INCLUDE_LIBC_INTERFACE})
-    target_include_directories(${TARGET_NAME} SYSTEM PRIVATE ${INCLUDE_EXTERNAL_LIBS_INTERFACE})
-    if(("${PLATFORM}" STREQUAL "DARWIN") AND (NOT (CMAKE_COMPILER_IS_GNUCC)))
-     target_link_libraries(${TARGET_NAME} ${CORE_TARGET_NAME} ${LIBC_TARGET_NAME}
-                           ${LIBM_TARGET_NAME} ${PREFIX_IMPORTED_LIB}libclang_rt.osx)
-    else()
-     target_link_libraries(${TARGET_NAME} ${CORE_TARGET_NAME} ${LIBC_TARGET_NAME}
-                           ${LIBM_TARGET_NAME} ${PREFIX_IMPORTED_LIB}libgcc)
-    endif()
+# C
+set(C_FLAGS_COMMON "-std=c99 -pedantic")
 
-    if("${PLATFORM}" STREQUAL "MCU")
-     add_dependencies(${TARGET_NAME} mcu_header_with_script_to_run.${TARGET_NAME})
-     add_custom_target(${TARGET_NAME}.bin
-                       COMMAND ${CMAKE_OBJCOPY} -Obinary $<TARGET_FILE:${TARGET_NAME}> $<TARGET_FILE:${TARGET_NAME}>.bin
-                       DEPENDS ${TARGET_NAME})
-     add_custom_target(${TARGET_NAME}.flash
-                       COMMAND st-flash write $<TARGET_FILE:${TARGET_NAME}>.bin 0x08000000
-                       DEPENDS ${TARGET_NAME}.bin)
-    endif()
+# Strip binary
+if(ENABLE_STRIP AND NOT CMAKE_BUILD_TYPE STREQUAL "Debug")
+  set(LINKER_FLAGS_COMMON "${LINKER_FLAGS_COMMON} -s")
+endif()
 
-    if("${BUILD_MODE}" STREQUAL "RELEASE")
-     if(${STRIP_RELEASE_BINARY} STREQUAL "ON")
-      add_custom_command(TARGET ${TARGET_NAME}
-                         POST_BUILD
-                         COMMAND ${CMAKE_STRIP} $<TARGET_FILE:${TARGET_NAME}>)
-     endif()
-    endif()
-   else()
-    add_custom_target(${TARGET_NAME} ALL)
-
-    add_custom_command(TARGET ${TARGET_NAME}
-                       POST_BUILD
-                       COMMAND mkdir -p ${CMAKE_BINARY_DIR}/${TARGET_NAME}
-                       COMMAND echo $<TARGET_FILE:${LIBM_TARGET_NAME}> > ${CMAKE_BINARY_DIR}/${TARGET_NAME}/list
-                       COMMAND echo $<TARGET_FILE:${CORE_TARGET_NAME}> >> ${CMAKE_BINARY_DIR}/${TARGET_NAME}/list)
-
-    if(DEFINED EXTERNAL_BUILD_ENTRY_FILE)
-     add_library(${TARGET_NAME}-entry STATIC ${EXTERNAL_BUILD_ENTRY_FILE})
-     set_property(TARGET ${TARGET_NAME}-entry
-                  PROPERTY COMPILE_FLAGS "${COMPILE_FLAGS_JERRY} ${FLAGS_COMMON_${BUILD_MODE}}")
-     target_compile_definitions(${TARGET_NAME}-entry PRIVATE ${DEFINES_JERRY})
-     target_include_directories(${TARGET_NAME}-entry PRIVATE ${INCLUDE_CORE_INTERFACE})
-     target_include_directories(${TARGET_NAME}-entry SYSTEM PRIVATE ${CMAKE_SOURCE_DIR})
-     target_include_directories(${TARGET_NAME}-entry SYSTEM PRIVATE ${INCLUDE_LIBC_INTERFACE})
-     target_include_directories(${TARGET_NAME}-entry SYSTEM PRIVATE ${INCLUDE_EXTERNAL_LIBS_INTERFACE})
-     add_dependencies(${TARGET_NAME} ${TARGET_NAME}-entry)
-    endif()
+# Jerry's libc
+if(BUILD_JERRY_LIBC STREQUAL "ON")
```

Comment:
No need for `STREQUAL`, `OFF` will automatically evaluate to false, and `ON` to true.